app/helpers/physical_switch_helper/textual_summary.rb

```diff
@@ -30,8 +30,13 @@ def textual_group_firmware_details
     TextualTable.new(_("Firmwares"), firmware_details, [_("Name"), _("Version")])
   end
 
-  def textual_group_ports
-    TextualTable.new(_("Ports"), port_details, [_("Name"), _("Type"), _("Peer MAC Address")])
+  def textual_ports
+    ports_count = @record.physical_network_ports.count
+    ports = {:label => _("Ports"), :value => ports_count, :icon => "ff ff-network-port"}
+    if ports_count.positive?
+      ports[:link] = "/physical_switch/show/#{@record.id}?display=ports"
+    end
+    ports
```

Comment:
You might want to take a look at `textual_link` that calss `textual_object_link` etc.

I might look complex at first, but you may find that it solves the issue of `.positive?`, calls a helper for constructing the URL and handles the case where the user does not have the permissions to follow the link.

Reply:
Well, I am not sure how to proceed with it right now.
Here https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/helpers/textual_summary_helper.rb#L171
we use `controller_for_model` and this will use the `base_model` for `physical_switch` which will return `switch` as a controller.

Reply:
Oh, I see. So we need to fix `controller_for_model` to work with switches and other stuff that it does not work with. Do you know any other cases that do not work?

You don't need to use it in this PR. I'm just trying to find ways to make the code more generic so that it's easier to work with and transform.

Reply:
> Do you know any other cases that do not work?

Not really, just stumbled on it as I was working on this PR, but if I do I will let you know.